### PR TITLE
Add factory for Orientation

### PR DESCRIPTION
**Pull request by Lignium**
Add factory for Orientation and improve SpongeOrientation class (hashCode, equals and toString).

### CODE REVIEW

File: src/main/java/org/spongepowered/common/util/SpongeOrientation.java [hunk 1]
```diff
@@ -26,16 +26,54 @@
...
     }
+
+    @Override
+    public int hashCode() {
+        return this.angle;
+    }
+
+    @Override
+    public boolean equals(final Object anotherObject) {
+        if (anotherObject == null) return false;
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/util/SpongeOrientation.java:66]: No shorthand ifs please.

File: src/main/java/org/spongepowered/common/util/SpongeOrientation.java [hunk 2]
```diff
@@ -26,16 +26,54 @@
...
         return this.angle;
     }
+
+    @Override
+    public int hashCode() {
+        return this.angle;
+    }
+
+    @Override
+    public boolean equals(final Object anotherObject) {
```
Comment (Zidane) [inline src/main/java/org/spongepowered/common/util/SpongeOrientation.java:65]: s/anotherObject/obj

File: src/main/java/org/spongepowered/common/util/SpongeOrientation.java (hunk 1 above)
Comment (Zidane) [reply src/main/java/org/spongepowered/common/util/SpongeOrientation.java:66]: Did not mean to resolve, instanceof check would handle the null check.

File: src/main/java/org/spongepowered/common/util/SpongeOrientation.java (hunk 2 above)
Comment (kashike) [reply src/main/java/org/spongepowered/common/util/SpongeOrientation.java:65]: or just "object" is fine too.

**Pull request closed without merging**